polaris-react/src/components/Button/Button.module.scss

```diff
@@ -398,4 +400,13 @@
     white-space: nowrap;
   }
 }
+
+[data-buttongroup-connected-top='true'] > *:first-child .Button {
+  border-top-left-radius: var(--p-border-radius-0);
+}
+
+[data-buttongroup-connected-top='true'] > *:last-child .Button {
+  border-top-right-radius: var(--p-border-radius-0);
+}
```

Comment:
I double checked the `ButtonGroup -- WithAllGaps` story because Chromatic was showing a [baseline change](https://shopify.chromatic.com/test?appId=5d559397bae39100201eedc1&id=65b83899efc73cc827092e57) that needed to be accepted for the first button group that had `connectedTop`. I confirmed that this matches an [older version](https://5d559397bae39100201eedc1-mpdoxqrfgo.chromatic.com/?path=/story/all-components-buttongroup--with-all-gaps) we had in storybook prior to the button refactoring.